src/metabase/query_processor/pivot.clj

```diff
@@ -39,20 +53,34 @@
 
     (group-bitmask 3 [1])   ; -> [_ 1 _] -> 101 -> 101 -> 5
     (group-bitmask 3 [1 2]) ; -> [_ 1 2] -> 100 -> 011 -> 1"
-  [num-breakouts indices]
+  [num-breakouts :- ::num-breakouts
+   indices       :- [:sequential ::index]]
   (transduce
    (map (partial bit-shift-left 1))
    (completing bit-xor)
    (int (dec (Math/pow 2 num-breakouts)))
    indices))
 
-(defn- breakout-combinations
+(mr/def ::breakout-combination
+  [:sequential ::index])
+
+(mr/def ::breakout-combinations
+  [:and
+   [:sequential ::breakout-combination]
+   [:fn
+    {:error/message "Distinct combinations"}
+    distinct?]])
```

Comment:
I think this will always return true, because `distinct?` is dumb  (careful of `(apply distinct? [])` too).
https://clojuredocs.org/clojure.core/distinct_q